Formula/n/nubesgen.rb

```diff
@@ -0,0 +1,28 @@
+class Nubesgen < Formula
+  desc "Kickstart your project on Azure in minutes!"
+  homepage "https://nubesgen.com/"
+  url "https://github.com/microsoft/NubesGen/releases/download/v0.17.0/nubesgen-cli-0.3.0-SNAPSHOT.jar"
+  version "0.17.0"
+  sha256 "6e8ee4fc4bc887f1741bcf758475f50f7cf0f706e91c74a80c541e7f9da2f06c"
+  license "MIT"
+
+  depends_on "azure-cli"
+  depends_on "gh"
+  depends_on "openjdk"
+
+  def install
+    libexec.install "nubesgen-cli-0.3.0-SNAPSHOT.jar" => "nubesgen-#{version}.jar"
+
+    # Force using the openjdk dependency
+    (bin/"nubesgen").write <<~EOS
+      #!/bin/bash
+      export JAVA_HOME="#{Language::Java.java_home_env[:JAVA_HOME]}"
+      exec "${JAVA_HOME}/bin/java" -jar "#{libexec}/nubesgen-#{version}.jar" "$@"
+    EOS
+  end
+
+  test do
+    s = pipe_output("#{bin}/nubesgen -h")
+    assert_match "CLI for NubesGen.com", s
+  end
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Indeed, we require authentication. But we also have a healthcheck command so I’ll see if we can use the strategy you suggest or the health check command

Reply:
I rewrote the test to use `healthcheck`. That being said, in order for the test to work whether the user is logged in or not to Azure, I'm only testing the presence of a string common to both the cases.

```
assert_match "Azure CLI is installed", s
```

Here is the log when the user is logged in:

```
NugesGen configuration starting
Checking required tools installation status...
Checking Azure CLI status...
Azure-cli    Azure-cli-core    Azure-cli-telemetry
-----------  ----------------  ---------------------
2.53.0       2.53.0            1.1.0
Azure CLI is installed.
EnvironmentName    HomeTenantId                          IsDefault    Name                                   State    TenantId
-----------------  ------------------------------------  -----------  -------------------------------------  -------  ------------------------------------
AzureCloud         *********************  True         ************** *****************
You are authenticated with Azure CLI.
GitHub CLI status...
gh version 2.36.0 (2023-10-03)
https://github.com/cli/cli/releases/tag/v2.36.0
GitHub CLI is installed.
github.com
  ✓ Logged in to github.com as ********* (keyring)
  ✓ Git operations for github.com configured to use ssh protocol.
  ✓ Token: ************************************
  ✓ Token scopes: ****, *****
You are authenticated with GitHub CLI.
```

And the log when the user is not logged in:

```
NugesGen configuration starting
Checking required tools installation status...
Checking Azure CLI status...
Azure-cli    Azure-cli-core    Azure-cli-telemetry
-----------  ----------------  ---------------------
2.53.0       2.53.0            1.1.0
Azure CLI is installed.
You are not authenticated with Azure CLI. Please run 'az login' first.
```

Let me know if that works

Reply:
Another test I can think of is the following

```
    s = pipe_output("#{bin}/nubesgen download")
    assert_predicate testpath/"terraform/README.md", :exist?
    assert_predicate testpath/"terraform/main.tf", :exist?
```

But I'm reluctant to use it as it depends on a template archive file downloaded from the https://nubesgen.com/ website and could break this homebrew formula for no reason if the template archive is modified server side.